distributed/scheduler.py

```diff
@@ -7471,7 +7472,11 @@ def _reevaluate_occupancy_worker(state: SchedulerState, ws: WorkerState):
 @cfunc
 @exceptval(check=False)
 def decide_worker(
-    ts: TaskState, all_workers, valid_workers: set, objective
+    ts: TaskState,
+    all_workers: set,
+    valid_workers: Optional[set],
+    idle: set,
+    objective,
```

Comment:
Everything here seems fine cython-wise, but pinging @jakirkham just in case

Reply:
There's one issue. Commented inline here ( https://github.com/dask/distributed/pull/4920/files#r652002105 )